src/main/scala/com/fulcrumgenomics/util/ReadStructure.scala

```diff
@@ -150,24 +141,64 @@ object ReadSegment {
     }
   }
 
-
   /** Creates a new read segment of the same type but with a new length */
   def apply(segment: ReadSegment, length: Int): ReadSegment = segment match {
     case seg: Template         => Template(seg.offset, length)
     case seg: SampleBarcode    => SampleBarcode(seg.offset, length)
     case seg: MolecularBarcode => MolecularBarcode(seg.offset, length)
     case seg: Skip             => Skip(seg.offset, length)
   }
+
+  def structureRead(segments: Seq[ReadSegment], bases: String, useOffset: Boolean = false): String = {
```

Comment:
I am not in love with this.  The reason for this method and its `useOffset` parameter, is if we the sample barcode as one contiguous sequence specified in the sample sheet, but the sample barcode is in multiple places across a given read (say inline at the start and end of read one).  I think an early commit in `ReadStructure` I had `ReadStructure. structureRead ` be able to only use a specific type of `ReadSegment`, which would be ideal.  Thoughts on how I can implement this more cleanly?